app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -236,13 +244,28 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
     }
 
     protected open fun startupTelemetryOnCreateCalled(safeIntent: SafeIntent, hasSavedInstanceState: Boolean) {
-        components.appStartupTelemetry.onHomeActivityOnCreate(safeIntent, hasSavedInstanceState)
+        components.appStartupTelemetry.onHomeActivityOnCreate(
+            safeIntent,
+            hasSavedInstanceState, homeActivityInitTimeStampNanoSeconds
+        )
+
+        OneShotPreDrawListener.add(rootContainer) {
+            components.appStartupTelemetry.onPreDraw()
+        }
     }
 
     override fun onRestart() {
         super.onRestart()
 
+        startupTelemetryOnRestartCalled()
+    }
+
+    private fun startupTelemetryOnRestartCalled() {
         components.appStartupTelemetry.onHomeActivityOnRestart()
+
+        OneShotPreDrawListener.add(rootContainer) {
```

Comment:
Same: can we call this in `onRestart`?

Reply:
see https://github.com/mozilla-mobile/fenix/pull/14375#discussion_r486829128